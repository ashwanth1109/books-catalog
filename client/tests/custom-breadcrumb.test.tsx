import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import CustomBreadcrumb from "../components/custom-breadcrumb.component";

const TITLE = "book-title";

describe("custom-breadcrumb.component.tsx", () => {
  it("should assert that no props breadcrumb shows only one level", () => {
    const { container } = render(<CustomBreadcrumb />);
    expect(screen.getByText("Books")).toBeInTheDocument();
    expect(() => {
      screen.getByText(TITLE);
    }).toThrowError();
    expect(container).toMatchSnapshot();
  });

  it("should assert that breadcrumbs has two levels with title", () => {
    const navFn = jest.fn();
    const { container } = render(
      <CustomBreadcrumb navigate={navFn} title={TITLE} />
    );
    expect(screen.getByText("Books")).toBeInTheDocument();
    expect(screen.getByText(TITLE)).toBeInTheDocument();
    expect(navFn).not.toBeCalled();
    fireEvent.click(screen.getByText("Books"));
    expect(navFn).toBeCalled();
    expect(container).toMatchSnapshot();
  });
});
