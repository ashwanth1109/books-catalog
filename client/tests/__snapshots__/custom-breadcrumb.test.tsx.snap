// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-breadcrumb.component.tsx should assert that breadcrumbs has two levels with title 1`] = `
<div>
  <div
    class="ant-breadcrumb"
    style="margin: 16px 0px;"
  >
    <span>
      <span
        class="ant-breadcrumb-link"
      >
        <span
          class="cursor-pointer"
        >
          Books
        </span>
      </span>
      <span
        class="ant-breadcrumb-separator"
      >
        /
      </span>
    </span>
    <span>
      <span
        class="ant-breadcrumb-link"
      >
        book-title
      </span>
      <span
        class="ant-breadcrumb-separator"
      >
        /
      </span>
    </span>
  </div>
</div>
`;

exports[`custom-breadcrumb.component.tsx should assert that no props breadcrumb shows only one level 1`] = `
<div>
  <div
    class="ant-breadcrumb"
    style="margin: 16px 0px;"
  >
    <span>
      <span
        class="ant-breadcrumb-link"
      >
        <span
          class="cursor-pointer"
        >
          Books
        </span>
      </span>
      <span
        class="ant-breadcrumb-separator"
      >
        /
      </span>
    </span>
  </div>
</div>
`;
